doc/source/whatsnew/v1.3.3.rst

```diff
@@ -21,6 +21,15 @@ Fixed regressions
 
 .. ---------------------------------------------------------------------------
 
+.. _whatsnew_133.performance:
+
+Performance improvements
+~~~~~~~~~~~~~~~~~~~~~~~~
+- Performance improvement for :meth:`DataFrame.__setitem__` with dataframes containing unique columns (:issue:`43274`)
```

Comment:
This is not specific enough. If key is any list-like or frame we don't hit this at all. Key has to be a scalar I think and value is not a df

Reply:
Updated. let me know how this looks